rmw_opensplice_cpp/src/guid.hpp

```diff
@@ -17,133 +17,19 @@
 
 #include <dds_dcps.h>
 
-#include <cstring>
-#include <iostream>
-#include <map>
-#include <set>
-#include <string>
-#include <vector>
 
-// TODO(ross-desmond): This should all be in opensplice
-typedef char octet;
-
-/**
- * Structure to hold GUID information for DDS instances.
- * http://www.eprosima.com/docs/fast-rtps/1.6.0/html/_guid_8h_source.html
- *
- */
-struct GuidPrefix_t
-{
-  static constexpr unsigned int kSize = 12;
-  octet value[kSize];
-
-  GuidPrefix_t()
-  {
-    memset(value, 0, kSize);
-  }
-
-  explicit GuidPrefix_t(octet guid[kSize])
-  {
-    memcpy(value, guid, kSize);
-  }
-
-  GuidPrefix_t(const GuidPrefix_t & g)
-  {
-    memcpy(value, g.value, kSize);
-  }
-
-  GuidPrefix_t(GuidPrefix_t && g)
-  {
-    memmove(value, g.value, kSize);
-  }
-
-  GuidPrefix_t & operator=(const GuidPrefix_t & guidpre)
-  {
-    memcpy(value, guidpre.value, kSize);
-    return *this;
-  }
-
-  GuidPrefix_t & operator=(GuidPrefix_t && guidpre)
-  {
-    memmove(value, guidpre.value, kSize);
-    return *this;
-  }
-
-#ifndef DOXYGEN_SHOULD_SKIP_THIS_PUBLIC
-
-  bool operator==(const GuidPrefix_t & prefix) const
-  {
-    return memcmp(value, prefix.value, kSize) == 0;
-  }
-
-  bool operator!=(const GuidPrefix_t & prefix) const
-  {
-    return memcmp(value, prefix.value, kSize) != 0;
-  }
-
-#endif
-};
-
-inline bool operator<(const GuidPrefix_t & g1, const GuidPrefix_t & g2)
-{
-  for (uint8_t i = 0; i < GuidPrefix_t::kSize; ++i) {
-    if (g1.value[i] < g2.value[i]) {
-      return true;
-    } else if (g1.value[i] > g2.value[i]) {
-      return false;
-    }
-  }
-  return false;
-}
-
-inline std::ostream & operator<<(std::ostream & output, const GuidPrefix_t & guiP)
-{
-  output << std::hex;
-  for (uint8_t i = 0; i < GuidPrefix_t::kSize - 1; ++i) {
-    output << static_cast<int>(guiP.value[i]) << ".";
-  }
-  output << static_cast<int>(guiP.value[GuidPrefix_t::kSize - 1]);
-  return output << std::dec;
-}
-
-// TODO(ross-desmond): check this with opensplice source code to ensure compatibility
-/**
- * Convert an instance handle to a guid.
- *
- * @param guid [out] the resulting guid
- * @param domain_id to prepend to the guid
- * @param instance_handle to append to the guid
- */
-inline void DDS_InstanceHandle_to_GUID(
-  struct GuidPrefix_t * guid,
-  DDS::DomainId_t domain_id,
-  DDS::InstanceHandle_t instance_handle)
-{
-  memcpy(guid->value, reinterpret_cast<char *>(&domain_id), sizeof(DDS::DomainId_t));
-  memcpy(guid->value + sizeof(DDS::DomainId_t),
-    reinterpret_cast<char *>(&instance_handle), sizeof(DDS::InstanceHandle_t));
-}
-
-inline void DDS_BuiltinTopicKey_to_GUID(
-  struct GuidPrefix_t * guid,
+inline void DDS_BuiltinTopicKey_to_InstanceHandle(
```

Comment:
Instead of having this take in an (output, input) parameter, I think it makes more sense to have this prototype be:

```
inline DDS::InstanceHandle_t DDS_BuiltinTopicKey_to_InstanceHandle(DDS::BuiltinTopicKey_t builtinTopicKey)
```

Then it is just a pure function.
```